05_classify_mutations_multimodal/nbconverted/compare_bayes_opt.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import matplotlib.patches as mpatches
+import seaborn as sns
+from adjustText import adjust_text
+
+import mpmp.config as cfg
+import mpmp.utilities.analysis_utilities as au
+import mpmp.utilities.plot_utilities as plu
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+
+# In[2]:
+
+
+grid_results_dir = Path(
+    cfg.results_dirs['multimodal'],
+    'compressed_shuffle_cancer_type',
+    'gene'
+).resolve()
+
+grid_unimodal_results_dir = Path(
+    cfg.results_dirs['mutation'],
+    'methylation_results_shuffle_cancer_type',
+    'gene'
+)
+
+bayes_opt_results_dir = Path(
+    cfg.results_dirs['multimodal'],
+    'bayes_opt',
+    'gene'
+).resolve()
+
+
+# In[3]:
+
+
+# load raw data
+grid_results_df = au.load_stratified_prediction_results(grid_results_dir, 'gene')
+
+# drop TET2 for now
+grid_results_df = grid_results_df[~(grid_results_df.identifier == 'TET2')]
+grid_results_df['param_opt'] = 'grid'
+
+# make sure that we have data for all data types and for two replicates (random seeds)
+print(grid_results_df.shape)
+print(grid_results_df.seed.unique())
+print(grid_results_df.identifier.unique())
+print(grid_results_df.training_data.unique())
+grid_results_df.head()
+
+
+# In[4]:
+
+
+# load expression and me_27k results
+u_results_df = au.load_compressed_prediction_results(grid_unimodal_results_dir, 'gene')
+genes = grid_results_df.identifier.unique()
+u_results_df = u_results_df[(u_results_df.n_dims == 5000) &
+                            (u_results_df.identifier.isin(genes))].copy()
+u_results_df.drop(columns='n_dims', inplace=True)
+u_results_df['param_opt'] = 'grid'
+
+# make sure data loaded matches our expectations
+print(u_results_df.shape)
+print(u_results_df.seed.unique())
+print(u_results_df.identifier.unique())
+print(u_results_df.training_data.unique())
+u_results_df.head()
+
+
+# In[5]:
+
+
+grid_results_df = pd.concat((
+    grid_results_df, u_results_df
+))
+group_df = (grid_results_df
+    .groupby(['identifier', 'training_data'])
+    .count()
```

Comment:
Depending on what you're looking for, it may be more helpful to call `unique` after `count`. In theory only one row should show up, indicating that all the fields for all the data types have the same counts. If that's not the case it should show you which data types/genes are different

Reply:
Good idea! That's a much more concise way to find the rows that are different.